test/integration/connectors/sql/test_postgres.py

```diff
@@ -1,21 +1,97 @@
 import tempfile
+from contextlib import contextmanager
 from pathlib import Path
 
+import faker
 import pandas as pd
 import pytest
 from psycopg2 import connect
 
-from test.integration.connectors.utils.constants import DESTINATION_TAG, env_setup_path
+from test.integration.connectors.utils.constants import DESTINATION_TAG, SOURCE_TAG, env_setup_path
 from test.integration.connectors.utils.docker_compose import docker_compose_context
+from test.integration.connectors.utils.validation import (
+    ValidationConfigs,
+    source_connector_validation,
+)
 from unstructured_ingest.v2.interfaces import FileData
 from unstructured_ingest.v2.processes.connectors.sql.postgres import (
     CONNECTOR_TYPE,
     PostgresAccessConfig,
     PostgresConnectionConfig,
+    PostgresDownloader,
+    PostgresDownloaderConfig,
+    PostgresIndexer,
+    PostgresIndexerConfig,
     PostgresUploader,
     PostgresUploadStager,
 )
 
+faker = faker.Faker()
+
+SEED_DATA_ROWS = 40
+
+
+@contextmanager
+def postgres_download_setup() -> None:
+    with docker_compose_context(docker_compose_path=env_setup_path / "sql" / "postgres" / "source"):
+        connection = connect(
+            user="unstructured",
+            password="test",
+            dbname="test_db",
+            host="localhost",
+            port=5433,
+        )
+        with connection.cursor() as cursor:
+            for _ in range(SEED_DATA_ROWS):
+                sql_statment = (
+                    f"INSERT INTO cars (brand, price) VALUES "
+                    f"('{faker.word()}', {faker.random_int()})"
+                )
+                cursor.execute(sql_statment)
+            connection.commit()
+        yield
+
+
+@pytest.mark.asyncio
+@pytest.mark.tags(CONNECTOR_TYPE, SOURCE_TAG, "sql")
+async def test_postgres_source():
+    connect_params = {
+        "host": "localhost",
+        "port": 5433,
+        "database": "test_db",
+        "user": "unstructured",
+        "password": "test",
+    }
+    with postgres_download_setup():
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Eventually would love to have this configurable to go to TempDirectory or a specified directory like in the .sh ingest tests. Nice to have.

Reply:
Why would we want that? This is a cleaner version of that, making sure the data is always cleanup up since for the test, we never need the contents that are written here.

Reply:
I've found it valuable to see what is being downloaded from source connectors in the past. I like to look at the .json files from the indexer and the .txt files from the downloader to troubleshoot. I can always just change it from tempdir to a named folder when I need to so not a big deal. But I always tell new connector devs to run the tests and then look at the resulting files as an example of what their "new" connector needs to be doing in the many phases.